taar/recommenders/similarity_recommender.py

```diff
@@ -0,0 +1,153 @@
+
+        donor_set = []
+        donor_index = 0
+
+        # This could be optimized by using cdist, however of the early
+        for donor in self.donors_list:
+            # Compute the similarity score between the donor and the client in the space of categorical features.
+            try:
+                # Here a larger distance indicates a poorer match between categorical variables.
+                j_d = (distance.hamming(client_categorical_feats, donor['categorical_features']))
+            except ValueError:
+                logger.error("Unable to compute similarity over categorical features",
+                             extra={"client_id": client_data["client_id"]})
+            # Compute the similarity score between the donor and the client in the space of numerical features.
+            try:
+                # Here a value close to zero indicates a good match in continuous variables.
+                j_c = (distance.canberra(client_continuous_feats, donor['continuous_features']))
+            except ValueError:
+                logger.error("Unable to compute similarity over continuous features for {}"
+                             .format(client_data["client_id"]))
+
+            # Take the product of similarities to attain a univariate similarity score.
+            # Addition of 0.001 to j_c avoids a zero value from the categorical variables, allowing j_d precedence
+            single_score = abs((j_c + 0.01) * j_d)
+
+            # compute the LR based on precomputed distributions that relate the score
+            #  to a probability of providing good addon recommendations
+            lr_from_score = self.get_lr(single_score)
+            donor_set.append((lr_from_score, donor_index))
+            donor_index += 1
+
+        return sorted(donor_set, reverse=True)
+
+    def recommend(self, client_data, limit=10):
+        donor_set_ranking = self.compute_similarity_to_donors(client_data)
+        # 2.0 corresponds to a likelihood ratio of 2 meaning that donors are less than twice
+        # as likely to be 'good'. A value > 1.0 is sufficient, but we like this to be high.
+        if donor_set_ranking[0][0] < 2.0:
+            logger.warning("Addons recommended with very low similarity score, perhaps donor set is unrepresentative",
+                           extra={"maximum_similarity": donor_set_ranking[0]})
+
+        recommendations = []
+        for donor_ranking in donor_set_ranking:
+            # This expects a list of lists, if singletons are included, they should single elements lists.
+            if donor_ranking[0] > 1.0:
+                # Retrieve the index of the highest ranked donors and then append their installed addons
+                recommendations.extend(self.donors_list[donor_ranking[1]]['active_addons'])
+            if len(recommendations) > limit:
+                return recommendations[:limit]
```

Comment:
nit: feels weird to have the same statement here :D instead of returning, let's just `break`